src/qt/importdialog.cpp

```diff
@@ -298,5 +344,377 @@ void ImportDialog::on_Accept_clicked() {
             }
             break;
         }
+        case importMulti: {
+            bool fRescan = true;
+            if (!ui->rescanButtonIA->isChecked())
+                fRescan = false;
+
+            if (fRescan && walletModel->wallet().chain().havePruned()) {
+                // Exit early and print an error.
+                // If a block is pruned after this check, we will import the key(s),
+                // but fail the rescan with a generic error.
+                QMessageBox::critical(this, tr("Rescan failed"), tr("Rescan is disabled when blocks are pruned"));
+                break;
+            }
+
+            wallet::WalletRescanReserver reserver = walletModel->wallet().getReserver();
+            if (fRescan && !reserver.reserve()) {
+                QMessageBox::critical(this, tr("Rescan failed"),
+                                      tr("Wallet is currently rescanning. Abort existing rescan or wait."));
+                break;
+            }
+
+            int64_t now = 0;
+            bool fRunScan = false;
+            int64_t nLowestTimestamp = 0;
+            std::vector<resultpdi> response = {};
```

Comment:
In c71f418f18000f537871f57a42ec76e0a25941e9 "qt: implemented importmulti and importdescriptors GUI's"

```suggestion
            std::vector<resultpdi> response;
```

Reply:
done